### PR TITLE
DOC: Clarify Exponent Order in ltisys.py

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

### CODE REVIEW

File: scipy/signal/ltisys.py [hunk 1]
```diff
@@ -1086,6 +1096,10 @@ def lsim2(system, U=None, T=None, X0=None, **kwargs):
     given to `lsim2` are passed on to `odeint`.  See the documentation
     for `scipy.integrate.odeint` for the full list of arguments.
 
+    If (num, den) is passed in for `system`, coefficients for both the
```
Comment (rkern) [inline scipy/signal/ltisys.py]: Either `system` or `*system` would be fine, but be consistent between the uses.

Comment (gfyoung) [inline scipy/signal/ltisys.py]: I actually was trying to match it to the argument signatures and current documentation, some of which use ```*system``` and others which use ```system```.  Should I just use ```system``` for all of them?

File: scipy/signal/ltisys.py [hunk 2]
```diff
@@ -375,6 +376,10 @@ class lti(object):
     of one of its subclasses: `StateSpace`, `TransferFunction` or
     `ZerosPolesGain`.
 
+    If a `TransferFunction` is passed in for `*system`, coefficients for both
+    the numerator and denominator should be specified in descending exponent
+    order (e.g. s^2 + 3s + 5 would be represented as [1, 3, 5]).
+
```
Comment (rkern) [inline scipy/signal/ltisys.py]: You aren't passing in a `TransferFunction` object: rather you are passing in numerator and denominator polynomials to *create* a `TransferFunction` object.

File: scipy/signal/ltisys.py (hunk 1 above)
Comment (rkern) [inline scipy/signal/ltisys.py]: Oh, I see, it's different for this function. That's fine, then.

File: scipy/signal/ltisys.py [hunk 3]
```diff
@@ -605,6 +610,11 @@ class TransferFunction(lti):
     state-space matrices) is very inefficient and may lead to numerical
     inaccuracies.
 
+    If a `TransferFunction` is passed in for `*system`, coefficients for
+    both the numerator and denominator should be specified in descending
+    exponent order (e.g. s^2 + 3s + 5 would be represented as [1, 3, 5]).
+    This specification also applies to the "array_like" argument option.
```
Comment (rkern) [inline scipy/signal/ltisys.py]: This statement only applies to the `(numerator, denominator)` tuple, not the `TransferFunction` object.

Comment (gfyoung) [inline scipy/signal/ltisys.py]: Ah, right.  Will change.

File: scipy/signal/ltisys.py (hunk 2 above)
Comment (gfyoung) [inline scipy/signal/ltisys.py]: Whoops, missed a copy/paste there.  Will fix as well.

**Comment by gfyoung** [comment]
@rkern : Updated with a happy Travis

**Comment by gfyoung** [comment]
Is there anything else that needs to be done here?  I've just been rebasing this PR onto ```master``` (which is why I keep triggering Travis builds).  Otherwise, I think this should be good to merge.

File: scipy/signal/ltisys.py [hunk 4]
```diff
@@ -375,6 +376,10 @@ class lti(object):
     of one of its subclasses: `StateSpace`, `TransferFunction` or
     `ZerosPolesGain`.
 
+    If (numerator, denominator) is passed in for `*system`, coefficients for
```
Comment (rgommers) [inline scipy/signal/ltisys.py]: This needs double backticks around both ``*system`` to render correctly in html/pdf docs (single backticks is for links to other objects/functions).

File: scipy/signal/ltisys.py [hunk 5]
```diff
@@ -375,6 +376,10 @@ class lti(object):
...
+    both the numerator and denominator should be specified in descending
+    exponent order (e.g. s^2 + 3s + 5 would be represented as [1, 3, 5]).
```
Comment (rgommers) [inline scipy/signal/ltisys.py]: Also double backticks for ``s^2 + 3s + 5``

File: scipy/signal/ltisys.py (hunk 4 above)
Comment (rgommers) [inline scipy/signal/ltisys.py]: same applies to changes in the rest of this file

**Comment by rgommers** [comment]
These look like good clarifications to me. Good to merge modulo the reST markup that I commented on.

**Comment by gfyoung** [comment]
@rgommers : Should the list ``[1, 3, 5]`` be also double-backticked?

**Comment by rgommers** [comment]
> @rgommers : Should the list [1, 3, 5] be also double-backticked?

Less critical, but yes I have a slight preference for doing so. It's basically a trade-off between making the html output look nicer but the plain docstring a bit less readable in a terminal. ``s^2..`` is much clearer (also because ``^`` is a special symbol in reST).

**Comment by gfyoung** [comment]
@rgommers : Okay, I'll add the back-ticks to those as well.

File: scipy/signal/ltisys.py (hunk 5 above)
Comment (gfyoung) [inline scipy/signal/ltisys.py]: Done.

File: scipy/signal/ltisys.py (hunk 4 above)
Comment (gfyoung) [inline scipy/signal/ltisys.py]: Done.

**Comment by gfyoung** [comment]
@rgommers : Updated with a happy Travis.

**Comment by rgommers** [comment]
Thanks @gfyoung, merged.

**Pull request merged**